core/src/main/java/tachyon/worker/hierarchy/StorageDir.java

```diff
@@ -117,6 +128,7 @@ public void accessBlock(long blockId) {
     synchronized (mLastBlockAccessTimeMs) {
       if (containsBlock(blockId)) {
         mLastBlockAccessTimeMs.put(blockId, System.currentTimeMillis());
+        updateReferenceFrequency(blockId, REFERENCE_FREQUENCY_INCREMENT);
```

Comment:
Yes, you pass the previous frequency to addBlockId if the block is copied. I mean you can pass a static initial value 1 to addBlockId if it is an indeed new block. that is, addBlockId(long, long,bollean) calls addBlock(long,long,long,1,bollean), which then simply updates the frequency.